rotkehlchen/chain/ethereum/modules/thegraph/decoder.py

```diff
@@ -0,0 +1,184 @@
+        }
+
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+        """Processes events from The Graph's staking contract"""
+        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(context)
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_delegated(self, context: DecoderContext) -> DecodingOutput:
+        """Processes StakeDelegated event"""
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        stake_amount = hex_or_bytes_to_int(context.tx_log.data[:32])
+        stake_amount_norm = token_normalized_value(
+            token_amount=stake_amount,
+            token=self.token,
+        )
+        # identify and override the original stake Transfer event
+        for event in context.decoded_events:
+            if (
+                event.location_label == delegator and
+                event.address == CONTRACT_THEGRAPH_STAKING and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                initial_amount_norm = event.balance.amount
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.balance = Balance(amount=stake_amount_norm)
+                event.notes = f'Delegated {stake_amount_norm} GRT to indexer {indexer}'
+                event.counterparty = CPT_THEGRAPH
+
+                # also account for the GRT burnt due to delegation tax
+                tokens_burnt = initial_amount_norm - stake_amount_norm
+                if tokens_burnt > 0:
+                    burn_event = self.base.make_event_from_transaction(
+                        transaction=context.transaction,
+                        tx_log=context.tx_log,
+                        event_type=HistoryEventType.SPEND,
+                        event_subtype=HistoryEventSubType.FEE,
+                        asset=A_GRT,
+                        balance=Balance(amount=tokens_burnt),
+                        location_label=delegator,
+                        notes=f'{tokens_burnt} GRT has been burnt as delegation tax',
+                        counterparty=CPT_THEGRAPH,
+                        address=context.tx_log.address,
+                    )
+                    return DecodingOutput(event=burn_event)
+                break
+        return DEFAULT_DECODING_OUTPUT
```

Comment:
This helps readability since it's the end of a loop and python does not have curlies

```suggestion
                break

        return DEFAULT_DECODING_OUTPUT
```